client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcResultSetWithEncoding.java

```diff
@@ -0,0 +1,159 @@
+        server.installPlugin(new MemoryPlugin());
+        server.createCatalog("memory", "memory");
+
+        try (Connection connection = createConnection(); Statement statement = connection.createStatement()) {
+            statement.executeUpdate("CREATE SCHEMA memory.tiny");
+            statement.executeUpdate("CREATE TABLE memory.tiny.lineitem as SELECT * from tpch.tiny.lineitem");
+        }
+    }
+
+    protected TestingTrinoServer createTestingServer()
+    {
+        minio = Minio.builder().build();
+        minio.start();
+        minio.createBucket(BUCKET_NAME, true);
+
+        Map<String, String> filesystemProperties = new java.util.HashMap<>();
+        filesystemProperties.put("fs.s3.enabled", "true");
+        filesystemProperties.put("fs.location", "s3://" + BUCKET_NAME + "/segments");
+        filesystemProperties.put("fs.segment.encryption", "false");
+        filesystemProperties.put("s3.endpoint", minio.getMinioAddress());
+        filesystemProperties.put("s3.region", MINIO_REGION);
+        filesystemProperties.put("s3.aws-access-key", MINIO_ACCESS_KEY);
+        filesystemProperties.put("s3.aws-secret-key", MINIO_SECRET_KEY);
+        filesystemProperties.put("s3.path-style-access", "true");
+        return TestingTrinoServer.builder()
+                .setPlugins(ImmutableList.of(new FileSystemSpoolingPlugin()))
+                .setSpoolingManagerConfiguration(Optional.of(new FactoryConfiguration("filesystem", filesystemProperties)))
+                .addProperty("experimental.protocol.spooling.enabled", "true")
+                .addProperty("protocol.spooling.shared-secret-key", randomAES256Key())
+                .build();
+    }
+
+    private static String randomAES256Key()
+    {
+        return getEncoder().encodeToString(createRandomAesEncryptionKey().getEncoded());
+    }
+
+    @Test
+    public void testBigDataSet()
+            throws Exception
+    {
+        try (ConnectedStatement connectedStatement = newStatement()) {
+            // Create a new recording
+            // Start a JFR recording
+            Process startRecording = Runtime.getRuntime().exec("jcmd " + getProcessId() + " JFR.start name=MyRecording filename=myrecording.jfr");
+            startRecording.waitFor();
+
+            Stopwatch stopwatch = Stopwatch.createStarted();
+            try (ResultSet resultSet = connectedStatement.getStatement().executeQuery("SELECT l.*\n" +
+                    "FROM memory.tiny.lineitem l, TABLE(sequence(\n" +
+                    "                start => 1,\n" +
+                    "                stop => 200,\n" +
+                    "                step => 1)) n")) {
+                while (resultSet.next()) {
+                    String test = resultSet.getString(2);
+                }
+                // Stop the JFR recording
+                Process stopRecording = Runtime.getRuntime().exec("jcmd " + getProcessId() + " JFR.stop name=MyRecording");
```

Comment:
??

if you need JFR you can capture it programmatically

Reply:
Not in java 8. However I don't think we need this feature in the production code, this was just to aid in providing the benchmarks.

Reply:
Tests can run with a higher Java